swimster-api/models/Reservation.js

```diff
@@ -124,6 +122,74 @@ x
         const reservations = result.rows
         return reservations
     }
+
+    static async fetchReservationsFromUser({ user }) {
+        // fetches all reservations created by user
+        const result = await db.query(`
+            SELECT  reservations.id,
+                    reservations.date,
+                    reservations.start_time,
+                    reservations.end_time,
+                    reservations.guests,
+                    reservations.total,
+                    reservations.status,
+                    reservations.created_at,
+                    users.email,
+                    (
+                        SELECT hostUsers.email
+                        FROM users AS hostUsers
+                        WHERE hostUsers.id = (
+                            SELECT listings.host_id
+                            FROM listings
+                            WHERE listings.id = listing_id
+                        )
+                    ) AS "host_email",
+                    listings.id AS "listing_id",
+                    listings.title AS "listing_title",
+                    listings.address AS "listing_address",
+                    listings.images AS "listing_images"
+            FROM reservations
+            JOIN users ON users.id = reservations.user_id
+            JOIN listings ON listings.id = reservations.listing_id
+            WHERE user_id = (SELECT id FROM users WHERE email = $1)
```

Comment:
https://github.com/Facebook-University-Projects/Swimster/blob/0edc33cd7b9219f72f28fbb5898591d88700fe89/swimster-api/models/Reservation.js#L182-L189

Is there any way to optimize this subquery? Should I still use `user.id` because the logged in user is usually the one seeing their listings' reservations

Reply:
Assuming that you're looking for "give me a list of all reservations on all listings owned by me", I think the following also gets the job done?
```
SELECT
    ...
FROM listings
JOIN reservations ON listings.id = reservations.listing_id
JOIN users ON users.id = reservations.user_id
WHERE listings.host_id = $1
```

Reply:
I think `reserver_user_id` might be too verbose of a name, but in the returning values for the SQL query I l make sure to return them in aliases.

The query worked by the way, so it's all good. Thanks